app/Stores/DatagridStore.js

```diff
@@ -7,36 +8,46 @@ class DatagridStore extends EventEmitter {
     constructor(...args) {
         super(...args);
 
-        this.entries = [];
-        this.sortDir = null;
-        this.sortField = null;
+        this.data = Map({
+            pending: true,
+            entries: List(),
+            sortDir: null,
+            sortField: null
+        });
     }
 
     loadData(view) {
+        this.data = this.data.update('pending', v => true);
+        this.emitChange();
+
         var sortField = this.sortField || view.sortField() || 'id';
         var sortDir = this.sortDir || view.sortDir() || 'DESC';
 
         ApiRequester
             .getAll(view, 1, true, [], sortField, sortDir)
             .then(function(data) {
-                this.entries = data;
+                this.data = this.data.update('entries', (list) => {
+                    list = list.clear();
+                    data.forEach((entry) => {
+                        list = list.push(fromJS(entry));
+                    });
+
+                    return list;
+                });
+                this.data = this.data.update('pending', v => false);
                 this.emitChange();
             }.bind(this));
     }
 
     sort(args) {
-        this.sortDir = args.sortDir;
-        this.sortField = args.sortField;
+        this.data = this.data.update('sortDir', v => args.sortDir);
+        this.data = this.data.update('sortField', v => args.sortField);
 
         return this.loadData(args.view);
     }
 
     getState() {
-        return {
-            entries: this.entries,
-            sortDir: this.sortDir,
-            sortField: this.sortField
-        }
+        return { data: this.data };
```

Comment:
Because React need a real javascript object for the state. But I can do that in the Component only.